spec/system/consumer/split_checkout_spec.rb

```diff
@@ -207,10 +207,114 @@
         end
       end
 
-      describe "filling out delivery details" do
+      describe "filling out order details" do
         before do
           fill_out_details
           fill_out_billing_address
+          order.update(user_id: user.id)
+        end
+
+        context "billing address" do
+          describe "checking the default address box" do
+            before do
+              check "order_save_bill_address"
+              choose free_shipping.name
+            end
+
+            it "creates a new default bill address" do        
+              expect {
+                proceed_to_payment
+                order.reload
+                user.reload
+              }.to change {
+                order.bill_address&.address1
+              }.from(nil).to("Rue de la Vie, 77")
+              .and change {
+                order.customer.bill_address&.address1
```

Comment:
In the code before, you didn't actually check the customer record. And as it turns out, there is not customer record before the checkout. It gets created during checkout. The easiest thing we can do here:

```suggestion
                order.customer&.bill_address&.address1
```

Reply:
Hum, right. That makes sense. An excellent, practical example on what a [safe navigation operator](https://en.wikipedia.org/wiki/Safe_navigation_operator) - something I was not aware of.

All this "micro-optimisation" around the PR was definitively well worth, I appreciate that @mkllnk  :pray: 

Green build, back to code review.